src/main/java/com/zufar/onlinestore/payment/api/PaymentApi.java

```diff
@@ -10,12 +10,40 @@
 
 public interface PaymentApi {
 
-    PaymentDetailsWithTokenDto createPayment(CreatePaymentDto createPaymentDto) throws StripeException;
+    /**
+     * This method allows to create a payment object
+     *
+     * @param createPaymentDto the request dto to create a payment object
+     * @return PaymentDetailsWithTokenDto combines payment details and a payment token for payment processing on the front end side
+     * @throws StripeException this is the parent of all Stripe API exceptions, in this method, exceptions descendants related to the payment creation and processing
+     * */
+    PaymentDetailsWithTokenDto createPayment(final CreatePaymentDto createPaymentDto) throws StripeException;
```

Comment:
What happens if you remove 'final' from a method declaration ?
Can you change the reference to this object without 'final' ?

Reply:
For 'final' below the same question